hack/update-codegen.sh

```diff
@@ -4,18 +4,16 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-export GO111MODULE=on
+
 # If we run with -mod=vendor here, then generate-groups.sh looks for vendor files in the wrong place.
 export GOFLAGS=-mod=
 
-if [ -z "${GOPATH:-}" ]; then
-  export GOPATH
-  GOPATH=$(go env GOPATH)
-fi
-
 REPO_ROOT=$(dirname "${BASH_SOURCE[@]}")/..
 CODEGEN_PKG=${CODEGEN_PKG:-"${REPO_ROOT}/vendor/k8s.io/code-generator"}
 
+# shellcheck disable=SC1091
+source "${REPO_ROOT}"/vendor/knative.dev/hack/codegen-library.sh
+
```

Comment:
Not sure if this is needed at this point (maybe in a separate PR for clean up or so). 
Comparing the output with the other PR I have I dont see any diff.

Reply:
Here is the diff we have:
```
$ git diff 2312 --stat
 .golangci.yaml                                                  |  1 +
 hack/patches/006-version.patch                                  | 11 ++++++-----
 hack/update-codegen.sh                                          | 10 ++++++----
 test/eventinge2erekt/features/encryption.go                     |  5 ++---
 test/extensione2erekt/containersource_kafkachannel_ksvc_test.go |  5 ++---
 test/extensione2erekt/features/encryption.go                    | 15 +++++++--------
 test/extensione2erekt/kafkasource_ksvc_test.go                  |  5 ++---
 test/extensione2erekt/servicemesh/multitenant_test.go           |  2 +-
 test/extensione2erekt/source_kafkabroker_ksvc_test.go           | 11 +++++------
 vendor/knative.dev/reconciler-test/pkg/manifest/manifest.go     | 20 ++------------------
 10 files changed, 34 insertions(+), 51 deletions(-)
```

Reply:
This is need because otherwise locally the codegen is broken for people like me not using Go path anymore https://github.com/openshift-knative/serverless-operator/pull/2312#issuecomment-1753095189

Reply:
CI job is green, so that's the source of truth for me

Reply:
Let's discuss in the sig release meeting, not sure this happens to everyone.

Reply:
>CI job is green, so that's the source of truth for me
Tests passed in the other PR too, also tests green does not mean what implemented is ok in general.
Let's discuss later. :)

Reply:
I'm not sure what is not clear but let me iterate why we need this.

If you don't clone the repository in your GOPATH, the directory where the code is generated is `$GOPATH/src/<go_module_name>` (if it's not explicitly set it's `~/go/src/<go_module_name>`), that's why I wasn't getting any diff.

With this import, the script is setting up a temporary "GOPATH" and it generates the code in the correct place.

Hope this helps clarifying what this does

btw, I need this PR to move forward with https://github.com/openshift-knative/serverless-operator/pull/2310

Reply:
Reference to the original commit https://github.com/knative/hack/pull/33, hope this helps

Reply:
and usage https://github.com/knative/hack/blob/158898889ae8670c7e68e522e32cd6800e1729a0/codegen-library.sh#L29